src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,117 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+               
+        public static bool IsInvisibleOutsideAssemblyAtSymbolLevel(ISymbol symbol)
+        {
+            return SymbolIsPrivateOrInternal(symbol)
+                || SymbolIsProtectedInSealed(symbol);
+        }
+
+        public static bool SymbolIsPrivateOrInternal(ISymbol symbol)
+        {
+            var access = symbol.DeclaredAccessibility;
+            return access == Accessibility.Private
+                || access == Accessibility.Internal
+                || access == Accessibility.ProtectedAndInternal
+                || access == Accessibility.NotApplicable;
+        }
+
+        public static bool SymbolIsProtectedInSealed(ISymbol symbol)
+        {
+            var containgType = symbol.ContainingType;
+            if (containgType != null && containgType.IsSealed)
+            {
+                var access = symbol.DeclaredAccessibility;
+                return access == Accessibility.Protected
+                    || access == Accessibility.ProtectedOrInternal;
+            }
+
+            return false;
+        }
+
+        public static string GetMeaningfulParentName(SyntaxNode cur, SemanticModel model)
```

Comment:
It is getting a enclosing method or class that has a non-empty name, which is used to get more specific information into the warning message.